Products/ZenHub/services/ProcessConfig.py

```diff
@@ -73,29 +74,36 @@ def _filterDevice(self, device):
 
         return include
 
-    def _createDeviceProxy(self, device):
+    def _createDeviceProxy(self, device, proxy=None):
         procs = device.getMonitoredComponents(collector='zenprocess')
         if not procs:
             log.debug("Device %s has no monitored processes -- ignoring",
                       device.titleOrId())
             return None
 
-        proxy = CollectorConfigService._createDeviceProxy(self, device)
+        proxy = CollectorConfigService._createDeviceProxy(
+            self, device, proxy=proxy)
+
         proxy.configCycleInterval = self._prefs.processCycleInterval
 
         proxy.name = device.id
         proxy.lastmodeltime = device.getLastChangeString()
         proxy.thresholds = []
         proxy.processes = {}
         proxy.snmpConnInfo = device.getSnmpConnInfo()
-        devuuid = device.getUUID()
         for p in procs:
+            tags = {}
+
             # Find out which datasources are responsible for this process
             # if SNMP is not responsible, then do not add it to the list
             snmpMonitored = False
             for rrdTpl in p.getRRDTemplates():
-                if len(rrdTpl.getRRDDataSources("SNMP")) > 0:
+                for datasource in rrdTpl.getRRDDataSources("SNMP"):
                     snmpMonitored = True
+                    for datapoint in datasource.getRRDDataPoints():
+                        tags = datapoint.getTags(p)
```

Comment:
This loop will only get the tags from first datapoint for the last datasource since the break only exits the inner loop. Is that the desired behavior? Maybe a comment on what this is doing.

Reply:
I'll add a comment. This is the case in PingTask too. They both essentially ignore the datapoint configurations and assume a static list. So I didn't go to the trouble of changing the overall strategy of how their configs are built.